### PR TITLE
Feature/guards

### PR DESCRIPTION
**Pull request by whaeck**
Header guards were added to the r2 library to accelerate unit test compilation.

In addition to these changes: 
- the eigen library dependency now points to the online Eigen repository, tagged version 3.3.8 (released about a month ago)
- the RLMatrixCalculator now also has a test associated to it

### CODE REVIEW

**Comment by jlconlin** [comment]
Is this at a point where we can incorporate GitHub Actions?

**Comment by whaeck** [comment]
@jlconlin I thought about that (I actually prepared the file and was about to push it before I changed my mind). The rmatrix SpinGroup test stills fails because of the CWF, so we may want to comment out the failing tests before doing that?

**Comment by jlconlin** [comment]
> @jlconlin I thought about that (I actually prepared the file and was about to push it before I changed my mind). The rmatrix SpinGroup test stills fails because of the CWF, so we may want to comment out the failing tests before doing that?

I think we should get GitHub Actions started. We can comment out failing tests if needed.

You can't just include the yml file. You have to create the first action and then merge that into the appropriate branch.

File: cmake/develop_dependencies.cmake [hunk 1]
```diff
@@ -11,11 +11,12 @@ FetchContent_Declare( ENDFtk
     GIT_SHALLOW     TRUE
     )
 
-FetchContent_Declare( eigen-adapter
-    GIT_REPOSITORY  https://github.com/njoy/eigen-adapter
-    GIT_TAG         origin/master
+FetchContent_Declare( eigen
```
Comment (jlconlin) [inline cmake/develop_dependencies.cmake:14]: This makes me very happy.

File: src/resonanceReconstruction/rmatrix/BoundaryCondition.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#include <string>
+
+// other includes
+
+namespace njoy {
+namespace resonanceReconstruction {
+namespace rmatrix {
+
+  // boundary condition
+  using BoundaryCondition = double;
```
Comment (jlconlin) [inline src/resonanceReconstruction/rmatrix/BoundaryCondition.hpp:14]: An entire file just for this? Seems a bit overkill. Can't this be included elsewhere?

File: cmake/develop_dependencies.cmake (hunk 1 above)
Comment (whaeck) [reply cmake/develop_dependencies.cmake:14]: It did take a lot of testing to get it to work, isn't that right @nathangibson14 ?

**Review by jlconlin — DISMISSED** [review]
This does improve the compile time---and memory requirements---when compiling the tests in this repository. I think that some of the additional files are a bit overkill, but it does work. 

This does not cause any immediate issues with compiling in [RECONR](https://github.com/njoy/RECONR), but it also doesn't reduce compiling expense. I haven't made any changes in RECONR to take advantage of these changes.

**Comment by whaeck** [comment]
@jlconlin You won't see any compilation time improvement in RECONR since you still include the entire library - which you cannot avoid in your case. For now, the reduction in compilation time will be only visible when only pieces of r2 are included (e.g. for the unit tests and python bindings once we start work on those).

**Comment by jlconlin** [comment]
> @jlconlin You won't see any compilation time improvement in RECONR since you still include the entire library - which you cannot avoid in your case. For now, the reduction in compilation time will be only visible when only pieces of r2 are included (e.g. for the unit tests and python bindings once we start work on those).

Yeah, that's what I was afraid of.

**Comment by whaeck** [comment]
But it is a step in the right direction ;-)

**Review by jlconlin — APPROVED** [review]

**Comment by jlconlin** [comment]
I guess this needs to be done before the pedantisms.

**Pull request merged**